### PR TITLE
Goal: Create a single shared build environment.

### PR DESCRIPTION
**Pull request by renukamanavalan**
Brief:
./docker-build/build.sh -- Builds debian container with all packages required to build & test our LoM system. cd docker-build; ./run.sh -- Runs the container from the built image mounting your source dir (../) as lom-root You run make inside this container

Gain:
We all, including Azure-build-pipeline will use the same image, hence same container to build LoM binaries.

### CODE REVIEW

File: docker-build/lom-slave-bullseye/Dockerfile
Comment (prithviraghavtm) [inline docker-build/lom-slave-bullseye/Dockerfile]: Should the maintainer still be gulv@ ?

File: docker-build/lom-slave-bullseye/Dockerfile
Comment (prithviraghavtm) [inline docker-build/lom-slave-bullseye/Dockerfile:45]: Could we reduce this docker image size by removing non-needed dependencies ? 

Since we are not using any c or python dependencies currently, could we bring down the size of the image with debian and go runtime installed, instead of 5 GB docker image ? 

Debian image consumes around 125 MB and go runtime should take less than 100 MB.

Comment (prithviraghavtm) [reply docker-build/lom-slave-bullseye/Dockerfile:45]: The goals would still be met:
- We will adhere to azure build pipeline (just that non-needed dependencies are removed), but the same debian layer would be used underneath.
- We will have consistency across team for using same dev environment.

File: docker-build/lom-slave-bullseye/Dockerfile
Comment (prithviraghavtm) [inline docker-build/lom-slave-bullseye/Dockerfile:10]: Will we be copying the lom source code in our custom dockerfile ?

File: docker-build/lom-slave-bullseye/lom-slave-bullseye-admin.log
Comment (prithviraghavtm) [inline docker-build/lom-slave-bullseye/lom-slave-bullseye-admin.log:1]: Are these log files only internal to slave docker ?

File: docker-build/lom-slave-bullseye/Dockerfile
Comment (renukamanavalan) [reply docker-build/lom-slave-bullseye/Dockerfile]: ;)

File: docker-build/lom-slave-bullseye/Dockerfile
Comment (renukamanavalan) [reply docker-build/lom-slave-bullseye/Dockerfile:45]: No need. Said that, you can remove ones that are not needed. But no gain.

File: docker-build/lom-slave-bullseye/Dockerfile
Comment (renukamanavalan) [reply docker-build/lom-slave-bullseye/Dockerfile:10]: We mount the dir. No copying

File: docker-build/lom-slave-bullseye/lom-slave-bullseye-admin.log
Comment (renukamanavalan) [reply docker-build/lom-slave-bullseye/lom-slave-bullseye-admin.log:1]: yep

File: docker-build/lom-slave-bullseye/Dockerfile
Comment (prithviraghavtm) [reply docker-build/lom-slave-bullseye/Dockerfile:45]: Are we not gaining 4.5 GB if the docker can be spawned with .5 GB size ? If yes, we can make the changes here itself in the code ?

File: docker-build/lom-slave-bullseye/Dockerfile
Comment (renukamanavalan) [reply docker-build/lom-slave-bullseye/Dockerfile]: fixed.

File: docker-build/lom-slave-bullseye/Dockerfile
Comment (renukamanavalan) [reply docker-build/lom-slave-bullseye/Dockerfile:45]: This is slave docker that runs in your dev environment.
It is transient and you need it only when you run make.
So size does not matter.
Your VM has  2TB
Please make sure,  docker uses /data for its workspace.

File: docker-build/lom-slave-bullseye/Dockerfile.cleanup
Comment (renukamanavalan) [inline docker-build/lom-slave-bullseye/Dockerfile.cleanup:1]: drop .dockerfile.swp

File: docker-build/lom-slave-bullseye/Dockerfile
Comment (prithviraghavtm) [inline docker-build/lom-slave-bullseye/Dockerfile:1]: You can add a readme or add comments here on steps to use this dockerfile.

**Review by prithviraghavtm — APPROVED** [review]

File: docker-build/lom-slave-bullseye/Dockerfile
Comment (renukamanavalan) [reply docker-build/lom-slave-bullseye/Dockerfile:1]: As for Dockerfile, one may refer Docker hub for detailed doc.
I added a README in docker-build for overall.

File: docker-build/slave_docker_run.sh
Comment (GouthamBanala) [inline docker-build/slave_docker_run.sh:11]: Just suggestion to differentiate it if we have lot of containers : 

USERNAME=$(whoami)
CUSTOM_NAME="my-custom-name"
CONTAINER_NAME="lom-build-$USERNAME-$CUSTOM_NAME"

docker run 
...
 --name $CONTAINER_NAME
...

**Review by GouthamBanala — APPROVED** [review]
Added just one minor suggestion. 
Approving overall after I installed this container on my VM and login to it.

File: docker-build/lom-slave-bullseye/Dockerfile.cleanup
Comment (renukamanavalan) [reply docker-build/lom-slave-bullseye/Dockerfile.cleanup:1]: Done.

File: docker-build/slave_docker_run.sh
Comment (renukamanavalan) [reply docker-build/slave_docker_run.sh:11]: Just one slave container only.
The LoM container we build will be distinct by image name  & will be versioned too and it will not be loaded in your Dev VM by default

**Pull request merged**
